packages/back-end/src/jobs/updateExposureQueriesDimensionSlices.ts

```diff
@@ -0,0 +1,266 @@
+import Agenda, { Job } from "agenda";
+import { subDays } from "date-fns";
+import { logger } from "back-end/src/util/logger";
+import { getContextForAgendaJobByOrgId } from "back-end/src/services/organizations";
+import {
+  _dangerousGetAllDatasourcesWithExposureQueriesWithAutomaticDimensionSlices,
+  getDataSourceById,
+  updateDataSource,
+} from "back-end/src/models/DataSourceModel";
+import { usingFileConfig } from "back-end/src/init/config";
+import { getIntegrationFromDatasourceId } from "back-end/src/services/datasource";
+import {
+  createDimensionSlices,
+  _dangerousGetAllDimensionSlicesByIds,
+} from "back-end/src/models/DimensionSlicesModel";
+import { DimensionSlicesQueryRunner } from "back-end/src/queryRunners/DimensionSlicesQueryRunner";
+
+const QUEUE_EXPOSURE_QUERIES_DIMENSION_SLICES_UPDATES =
+  "queueExposureQueriesDimensionSlicesUpdates";
+
+const UPDATE_SINGLE_DATASOURCE_EXPOSURE_QUERIES_DIMENSION_SLICES =
+  "updateSingleExposureQueryDimensionSlices";
+
+const DEFAULT_LOOKBACK_DAYS = 30;
+
+type UpdateSingleJobParams = {
+  organizationId: string;
+  dataSourceId: string;
+  exposureQueryIds: string[];
+};
+
+export default async function (agenda: Agenda) {
+  agenda.define(QUEUE_EXPOSURE_QUERIES_DIMENSION_SLICES_UPDATES, async () => {
+    const exposureQueriesToUpdate =
+      await getDataSourcesWithExposureQueriesToUpdate();
+
+    for (const jobParams of exposureQueriesToUpdate) {
+      await queueExposureQueriesDimensionSlicesUpdate(jobParams);
+    }
+  });
+
+  agenda.define(
+    UPDATE_SINGLE_DATASOURCE_EXPOSURE_QUERIES_DIMENSION_SLICES,
+    updateSingleDatasourceExposureQueriesDimensionSlices,
+  );
+
+  // Don't schedule automatic updates if using file config
+  if (usingFileConfig()) return;
+
+  await startUpdateJob();
+
+  async function startUpdateJob() {
+    const job = agenda.create(
+      QUEUE_EXPOSURE_QUERIES_DIMENSION_SLICES_UPDATES,
+      {},
+    );
+    job.unique({});
+    job.repeatEvery("1 day");
```

Comment:
It should be every 7 days, and eventually maybe customizable.

Reply:
While we run the `queueUpdates` job every day, in the job we check for slices that have been updated more than 7 days ago ([this line / parameter](https://github.com/growthbook/growthbook/pull/4625/files#diff-ea9bb28b02ab66b3eb377aeebc006683e2f548470d98224c80aeac3574189fcfR116)).

So each individual dimensionSlice has their own 7 days clock, instead of us updating every single record every Saturday for example.

This will also allow us to have different schedules per user in the future.